controller.py

```diff
@@ -93,15 +96,19 @@ def registerAction(self, name: str, callback: Callable) -> None:
   def performAction(self, name: str, args: Optional[Tuple[any]]) -> None:
     if not name in self.actions:
       raise Exception("Actions with the name does not exist '", name, "' in: ", type(self).__name__)
-    
+
     if not callable(self.actions[name]):
       raise Exception("Registered action must be a function: ", name)
 
     self.actions[name](args=args)
-  
+
   """ Function called on controller shutdown
 
       To be overridden by classes implementing Controller if needed
   """
+  def run(self):
+      while(True):
```

Comment:
Typically in Python, brackets aren't used for loops/if statement unless you need to have complicated conditions.
Also, the sleep function is part of the time library, not threads. So it would need to be
`time.sleep(num)`, where `num` is the number of seconds. Could just leave it a 1 sec or something for now.

We would also need to make it be `while self.running` instead of `while True`, otherwise we will get stuck in an infinite loop.